kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/data/DataServiceImpl.java

```diff
@@ -623,8 +623,9 @@ public void run() {
                         // There's nothing we can do here but log an exception.
                         logger.error("Unexpected Error. Task will be terminated", e);
                         throw e;
+                    } catch (Exception e) {
```

Comment:
Not sure why we added this catch Exception here.
In theory, the RuntimeException caught in MqttDataTransport is thrown as KuraCOnnectException that should be catched in Dataservice at line 609.
Am I wrong on this?